libslide/src/evaluator_rules/registry.rs

```diff
@@ -37,6 +38,23 @@ define_rules! {
          UnwrapExplicitBrackets: S("[_a] -> _a")
 }
 
+impl PartialOrd for RuleName {
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+impl Ord for RuleName {
+    fn cmp(&self, other: &Self) -> Ordering {
+        if self == other {
+            return Ordering::Equal;
+        } else if (*self as u8) < (*other as u8) {
+            return Ordering::Less;
+        }
+        Ordering::Greater
+    }
```

Comment:
can't you just do *self as u8.cmp(*other as u8) here?

Reply:
smart...